lib/obj/src/data_segment.rs

```diff
@@ -0,0 +1,34 @@
+use faerie::{Artifact, Decl};
+use wasmtime_environ::DataInitializer;
+
+/// Declares data segment symbol
+pub fn declare_data_segment(
+    obj: &mut Artifact,
+    _data_initaliazer: &DataInitializer,
+    index: usize,
+) -> Result<(), String> {
+    let name = format!("_memory_{}", index);
+    obj.declare(
+        name,
+        Decl::Data {
+            writable: false,
+            global: false,
+        },
+    )
+    .map_err(|err| format!("{}", err))?;
+    Ok(())
+}
+
+/// Emit segment data and initialization location
+pub fn emit_data_segment(
+    obj: &mut Artifact,
+    data_initaliazer: &DataInitializer,
+    index: usize,
+) -> Result<(), String> {
+    let name = format!("_memory_{}", index);
+    // FIXME: We need to initialize memory in a way that supports alternate
```

Comment:
If this is a problem in faerie please file an issue over there about this :)

Reply:
I need to remove this comment -- it stayed from the past code.